sql/catalyst/src/test/scala/org/apache/spark/sql/types/DataTypeSuite.scala

```diff
@@ -876,6 +878,90 @@ class DataTypeSuite extends SparkFunSuite {
       }
   }
 
+  test("string field with invalid collation name") {
+    val collationProviders = Seq("spark", "icu")
+    collationProviders.foreach { provider =>
+      val json =
+        s"""
+           |{
+           |  "type": "struct",
+           |  "fields": [
+           |    {
+           |      "name": "c1",
+           |      "type": "string",
+           |      "nullable": false,
+           |      "metadata": {
+           |        "${DataType.COLLATIONS_METADATA_KEY}": {
+           |          "c1": "$provider.INVALID"
+           |        }
+           |      }
+           |    }
+           |  ]
+           |}
+           |""".stripMargin
+
+      // Check that the exception will be thrown in case of invalid collation name and
+      // UNKNOWN_COLLATION_NAME config not enabled.
+      checkError(
+        exception = intercept[SparkException] {
+          DataType.fromJson(json)
+        },
+        condition = "COLLATION_INVALID_NAME",
+        parameters = Map(
+          "proposals" -> "id",
```

Comment:
Probably, it is not related to the PR. Just wonder why `id` is among proposals.

Reply:
As far as I understand, the proposal parameter is used to propose the collation that is "closest" to the name that is provided here(closest means it differs in least number of characters) and here, the returned value is `id`. It is handled in `CollationFactory.getClosestSuggestionOnInvalidName` method.